src/app/util/binding-table.h

```diff
@@ -23,8 +23,67 @@
 
 #include <app/util/af-types.h>
 
-EmberStatus emberGetBinding(uint8_t index, EmberBindingTableEntry * result);
+namespace chip {
 
-EmberStatus emberSetBinding(uint8_t index, EmberBindingTableEntry * result);
+class BindingTable
+{
+    friend class Iterator;
 
-EmberStatus emberDeleteBinding(uint8_t index);
+public:
+    BindingTable();
+
+    class Iterator
+    {
+        friend class BindingTable;
+
+    public:
+        EmberBindingTableEntry & operator*() { return mTable->mBindingTable[mIndex]; }
+
+        const EmberBindingTableEntry & operator*() const { return mTable->mBindingTable[mIndex]; }
+
+        EmberBindingTableEntry * operator->() { return &(mTable->mBindingTable[mIndex]); }
+
+        const EmberBindingTableEntry * operator->() const { return &(mTable->mBindingTable[mIndex]); }
+
+        Iterator operator++() const;
+
+        bool operator==(const Iterator & rhs) { return mIndex == rhs.mIndex; }
+
+        bool operator!=(const Iterator & rhs) { return mIndex != rhs.mIndex; }
+
+        uint8_t GetIndex() { return mIndex; }
+
+    private:
+        BindingTable * mTable;
+        mutable uint8_t mPrevIndex;
+        mutable uint8_t mIndex;
+    };
+
+    CHIP_ERROR Add(const EmberBindingTableEntry & entry);
+
+    EmberBindingTableEntry & GetAt(uint8_t index);
+
+    Iterator RemoveAt(Iterator iter);
+
+    uint8_t Size() { return mSize; }
```

Comment:
Should document that this returns the number of bindings stored, not the total possible number that could be stored.